### ISSUE TITLE
UndoRedo's do/undo methods can't go further than 5 arguments

### ISSUE THREAD

**Issue by YeldhamDev** [godotengine/godot#26483]
**Godot version:**
b84b015

**Issue description:**
When using UndoRedo's do/undo methods, the number of arguments can't go further than 5, as anything else will be ignored.

**Minimal reproduction project:**
The following code will give an error stating that the `test()` method was only called with 5 arguments.
```
var _undo_redo := UndoRedo.new()

func _ready() -> void:
	_undo_redo.create_action("test")
	_undo_redo.add_do_method(self, "test",
			true, true, true, true, true, true, true)
	_undo_redo.commit_action()

func test(arg1: bool, arg2: bool, arg3: bool, arg4: bool, arg5: bool,
		arg6: bool, arg7: bool) -> void:
	print(arg1, arg2, arg3, arg4, arg5, arg6, arg7)
```

**Comment by Naryosha** [comment]
Limit is defined in `core/object.h`. Not sure why exactly 5.
https://github.com/godotengine/godot/blob/8b231b96e347b677ea4189784c960bc4517b8e6a/core/object.h#L45

**Comment by YeldhamDev** [comment]
@Naryosha Then why methods like `print()` and `call()` can surpass it?

**Comment by Naryosha** [comment]
Both functions don't refer to the `VARIANT_ARG_MAX` when looping through arguments. I think `Object::call` used to have the limit but it's overloaded by a newer method. Not sure. Git blame traces back to the beginning of time.

**Comment by YeldhamDev** [comment]
@akien-mga Is there a reason for the argument limit to be 5, or is just arbitrary?

**Comment by Valeryn4** [comment]
function uses VARIANT_ARG_LIST
its makros "void add_do_method(Object *p_object, const String &p_method, VARIANT_ARG_LIST);"

But I do not know if the engine can work with "va_list".
I do not know the methods that can do any number of arguments.

**Comment by Mantissa-23** [comment]
Just encountered this while developing a plugin. If it's not going to be fixed in the next update, then an error specific to exceeding VARIANT_ARG_MAX instead of just a generic `Method expected X arguments, but called with 5.` error would be valuable.

**Comment by akien-mga** [comment]
Fixed by #54188.
